app/src/main/java/com/github/studydistractor/sdp/createDistraction/CreateDistractionViewModel.kt

```diff
@@ -11,61 +12,134 @@ import kotlinx.coroutines.flow.StateFlow
-            return Tasks.forException(Exception("Description is too long"))
+        if ((!uiState.value.latitude.isNullOrBlank()) && (!uiState.value.longitude.isNullOrBlank())) {
+            Log.d("CreateDistractionViewModel", "Latitude: ${uiState.value.latitude}, Longitude: ${uiState.value.longitude}")
+            if (!validateLatitude() || !validateLongitude()) {
+                return Tasks.forException(Exception("Invalid latitude or longitude"))
+            }
+            return _createDistractionModel.createDistraction(
+                Distraction(
+                    name = uiState.value.name,
+                    description = uiState.value.description,
+                    lat = uiState.value.latitude?.toDouble(),
+                    long = uiState.value.longitude?.toDouble()
+                )
+            )
+        }
+        if ((uiState.value.latitude.isNullOrBlank()) xor (uiState.value.longitude.isNullOrBlank())){
+            return Tasks.forException(Exception("Please fill in both latitude and longitude"))
         }
 
-        if(uiState.value.name == "" || uiState.value.description == "") {
-            return Tasks.forException(Exception("Please fill in the blanks"))
+        return _createDistractionModel.createDistraction(
+            Distraction(
+                name = uiState.value.name,
+                description = uiState.value.description
+            )
+        )
+    }
+
+    /**
+     * Update the visibility of the location fields.
+     * @param isVisible: the new visibility
+     */
+    fun updateLocationFieldIsVisible(isVisible: Boolean) {
+        _uiState.update {
+            it.copy(
+                locationFieldIsVisible = isVisible
+            )
         }
+    }
+
+    /**
+     * Validates the latitude value stored in the current UI state.
+     *
+     * @return True if the latitude value is valid and false otherwise.
+     */
+    private fun validateLatitude(): Boolean {
+        val latitudeDouble = uiState.value.latitude?.toDoubleOrNull() ?: return false
+
+        if (latitudeDouble < CreateDistractionViewModelConstants.MIN_LATITUDE || latitudeDouble > CreateDistractionViewModelConstants.MAX_LATITUDE) return false
+
+        return true
+    }
+
+    /**
+     * Validates the longitude value stored in the current UI state.
+     *
+     * @return True if the longitude value is valid and false otherwise.
+     */
+    private fun validateLongitude(): Boolean {
+        val longitudeDouble = uiState.value.longitude?.toDoubleOrNull() ?: return false
+
+        if (longitudeDouble < CreateDistractionViewModelConstants.MIN_LONGITUDE || longitudeDouble > CreateDistractionViewModelConstants.MAX_LONGITUDE) return false
 
-        return _createDistractionModel.createDistraction(Distraction(
-            name = uiState.value.name,
-            description = uiState.value.description
-        ))
+        return true
     }
 }
```

Comment:
Wait, does this PR is adding other PRs a part from the Create event screen ? I don't know if it is a good idea to mix them up

Reply:
After some discussions, the situation has been cleared out